### PR TITLE
Add missing `agent.activation_method` mapping

### LINKED ISSUE #10551: `agent.activation_method` is not mapped in data streams
As part of #9622 , #9850 added code to parse `agent.activation_method` from metadata and mapped it in traces only, which is not enough. It should be mapped in all data streams that store raw signals.

Example error due to missing mapping entry:
```
{"log.level":"error","@timestamp":"2023-03-29T12:32:31.086Z","log.origin":{"file.name":"go-docappender@v0.1.0/appender.go","file.line":279},"message":"failed to index document (strict_dynamic_mapping_exception): mapping set to strict, dynamic introduction of [activation_method] within [agent] is not allowed","service.name":"apm-server","ecs.version":"1.6.0"}
```

### PR DESCRIPTION
**Pull request by carsonip**
## Motivation/summary

Add missing agent.activation_method mapping to `logs-apm.app`, `metrics-apm.app`, `logs-apm.error` and `metrics-apm.internal`

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [x] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

## How to test these changes

- Send events (metrics, logs, transactions, spans) with `agent.activation_method` metadata
- Check that the event is indexed properly with the mapped field `agent.activation_method`

## Related issues

#9622 
Fixes #10551

### CODE REVIEW

**Review by kruskall — APPROVED** [review]
LGTM!

File: apmpackage/apm/changelog.yml [hunk 1]
```diff
@@ -3,6 +3,9 @@
     - description: Store app logs into service-specific data streams
       type: enhancement
       link: https://github.com/elastic/apm-server/pull/10456
+    - description: Add missing `agent.activation_method` mapping to `logs-apm.app`, `metrics-apm.app`, `logs-apm.error` and `metrics-apm.internal`
```
Comment (kruskall) [inline apmpackage/apm/changelog.yml:6]: **question**: shouldn't this be in the 8.7.1 section ?

**Review by kruskall — APPROVED** [review]
LGTM! Just a minor comment

**Review by kruskall — APPROVED** [review]

**Comment by lahsivjar** [comment]
Tested successfully with 8.7.1

<details>
<summary>Input NDJSON</summary>

```
{"metadata": {"service": {"name": "test-service-10552","node": {"configured_name": "node-123"},"version": "5.1.3","environment": "staging","language": {"name": "ecmascript","version": "8"},"runtime": {"name": "node","version": "8.0.0"},"framework": {"name": "Express","version": "1.2.3"},"agent": {"name": "elastic-node", "activation_method": "some_activation_method","version": "3.14.0"}},"user": {"id": "123user", "username": "bar", "email": "bar@user.com"}, "labels": {"tag0": null, "tag1": "one", "tag2": 2}, "process": {"pid": 1234,"ppid": 6789,"title": "node","argv": ["node","server.js"]},"system": {"hostname": "prod1.example.com","architecture": "x64","platform": "darwin", "container": {"id": "container-id"}, "kubernetes": {"namespace": "namespace1", "pod": {"uid": "pod-uid", "name": "pod-name"}, "node": {"name": "node-name"}}},"cloud":{"account":{"id":"account_id","name":"account_name"},"availability_zone":"cloud_availability_zone","instance":{"id":"instance_id","name":"instance_name"},"machine":{"type":"machine_type"},"project":{"id":"project_id","name":"project_name"},"provider":"cloud_provider","region":"cloud_region","service":{"name":"lambda"}}}}
{"transaction": { "id": "1111111111111111", "trace_id": "0123456789abcdef0123456789abcdef", "name": "amqp receive", "parent_id": "abcdefabcdef01234567", "type": "messaging", "duration": 3, "timestamp": 1682582069000000, "span_count": { "started": 1 }, "context": {"message": {"queue": { "name": "new_users"}, "age":{ "ms": 1577958057123}, "headers": {"user_id": "1ax3", "involved_services": ["user", "auth"]}, "body": "user created", "routing_key": "user-created-transaction"}},"session":{"id":"sunday","sequence":123}}}
```
</details>

<details>
<summary>Output assertion</summary>

#### Query
```
GET traces-apm-default/_search?expand_wildcards=all
{
  "query": {
    "bool": {
      "filter": [
        { "term": { "agent.activation_method": "some_activation_method" }},
        { "term": { "service.name": "test-service-10552" }}
      ]
    }
  }
}
```

#### Result
```
{
  "took": 4,
  "timed_out": false,
  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 1,
      "relation": "eq"
    },
    "max_score": 0,
    "hits": [
      {
        "_index": ".ds-traces-apm-default-2023.04.27-000001",
        "_id": "Wg-2wYcBfLcKIcBu-85Y",
        "_score": 0,
        "_source": {
          "container": {
            "id": "container-id"
          },
          "kubernetes": {
            "node": {
              "name": "node-name"
            },
            "pod": {
              "uid": "pod-uid",
              "name": "pod-name"
            },
            "namespace": "namespace1"
          },
          "parent": {
            "id": "abcdefabcdef01234567"
          },
          "process": {
            "args": [
              "node",
              "server.js"
            ],
            "parent": {
              "pid": 6789
            },
            "pid": 1234,
            "title": "node"
          },
          "agent": {
            "name": "elastic-node",
            "activation_method": "some_activation_method",
            "version": "3.14.0"
          },
          "session": {
            "sequence": 123,
            "id": "sunday"
          },
          "data_stream.namespace": "default",
          "data_stream.type": "traces",
          "processor": {
            "name": "transaction",
            "event": "transaction"
          },
          "labels": {
            "tag1": "one"
          },
          "cloud": {
            "availability_zone": "cloud_availability_zone",
            "instance": {
              "name": "instance_name",
              "id": "instance_id"
            },
            "provider": "cloud_provider",
            "service": {
              "name": "lambda"
            },
            "machine": {
              "type": "machine_type"
            },
            "project": {
              "name": "project_name",
              "id": "project_id"
            },
            "region": "cloud_region",
            "account": {
              "name": "account_name",
              "id": "account_id"
            }
          },
          "observer": {
            "hostname": "deafd7cd783b",
            "type": "apm-server",
            "version": "8.7.1"
          },
          "trace": {
            "id": "0123456789abcdef0123456789abcdef"
          },
          "@timestamp": "2023-04-27T07:54:29.000Z",
          "numeric_labels": {
            "tag2": 2
          },
          "service": {
            "node": {
              "name": "node-123"
            },
            "environment": "staging",
            "framework": {
              "name": "Express",
              "version": "1.2.3"
            },
            "name": "test-service-10552",
            "runtime": {
              "name": "node",
              "version": "8.0.0"
            },
            "language": {
              "name": "ecmascript",
              "version": "8"
            },
            "version": "5.1.3"
          },
          "data_stream.dataset": "apm",
          "host": {
            "hostname": "node-name",
            "os": {
              "platform": "darwin"
            },
            "ip": [
              "180.129.57.151"
            ],
            "name": "node-name",
            "architecture": "x64"
          },
          "event": {
            "agent_id_status": "missing",
            "ingested": "2023-04-27T07:56:47Z",
            "outcome": "unknown"
          },
          "user": {
            "name": "bar",
            "id": "123user",
            "email": "bar@user.com"
          },
          "transaction": {
            "duration": {
              "us": 3000
            },
            "representative_count": 1,
            "name": "amqp receive",
            "id": "1111111111111111",
            "span_count": {
              "started": 1
            },
            "type": "messaging",
            "message": {
              "headers": {
                "User_id": [
                  "1ax3"
                ],
                "Involved_services": [
                  "user",
                  "auth"
                ]
              },
              "body": "user created",
              "routing_key": "user-created-transaction",
              "queue": {
                "name": "new_users"
              },
              "age": {
                "ms": 1577958057123
              }
            },
            "sampled": true
          },
          "timestamp": {
            "us": 1682582069000000
          }
        }
      }
    ]
  }
}
```
</details>

**Pull request merged**